src/ape/api/config.py

```diff
@@ -13,82 +12,29 @@ class ConfigEnum(str, Enum):
     """
 
 
-@dataclass(slots=True, kwargs=True)
-class ConfigItem:
+class ConfigDict(BaseModel):
+    __root__: dict = {}
+
+
+class PluginConfig(BaseSettings):
     """
     A base plugin configuration class. Each plugin that includes
     a config API must register a subclass of this class.
     """
 
-    def serialize(self) -> Dict:
-        """
-        Serialize the config item into a raw dict format for storing on disk.
-
-        Returns:
-            Dict
-        """
-        data: Dict[str, Union[str, int, Dict, List, None]] = dict()
-        for name in self.__slots__:
-            value = getattr(self, name)
-            if isinstance(value, ConfigItem):
-                data[name] = value.serialize()
-            elif isinstance(value, ConfigEnum):
-                data[name] = value.name
-            elif value is None or isinstance(value, (int, str, dict, list)):
-                data[name] = value
-            else:
-                logger.error(
-                    f"Received unknown type '{type(value)}' when serializing a config item."
-                )
-        return data
-
-    def validate_config(self):
-        pass
-
-    def __getitem__(self, attrname: str) -> Any:
-        """
-        Get a configuration setting property by name. Use
-        :meth:`~ape.api.config.ConfigItem.get` when it is ok for the key not to
-        exist in the config.
-
-        Raises:
-            KeyError: When the attribute name is not a key in the config.
-
-        Args:
-            attrname (str): The configuration setting key.
-
-        Returns:
-            Any: The value from the config.
-        """
-
-        if attrname in self.__slots__:
-            return getattr(self, attrname)
-
-        raise KeyError(f"{attrname!r}")
-
-    def get(self, attrname: str, default_value: Optional[Any] = None) -> Optional[Any]:
-        """
-        Get a configuration setting property by name.
-
-        Args:
-            attrname (str): The configuration setting key.
-            default_value (Optional[Any]): The value to return if the key does
-              not exist. Defaults to ``None``.
-
-        Returns:
-            Optional[Any]: The default value if the key is not in the config, the value otherwise.
-        """
+    def __getitem__(self, attr_name: str) -> Any:
+        # allow hyphens in plugin config files
+        attr_name = attr_name.replace("-", "_")
 
-        if attrname in self.__slots__:
-            return getattr(self, attrname)
+        if not hasattr(self, attr_name):
+            raise AttributeError(f"{self.__class__.__name__} has no attr '{attr_name}'")
 
-        return default_value
+        return getattr(self, attr_name)
 
 
-class ConfigDict(ConfigItem):
+class UnprocessedConfig(PluginConfig):
```

Comment:
I find the name of the class confusing... `Unprocessed` makes it sound like the it is completely ignored?

Reply:
Maybe one of these:

* `FreeformConfig`
* `GenericConfig`
* `AnyConfig`
* `UnspecializedConfig`

Reply:
I pulled this from @fubuloubu's original refactor. I'm leaning toward `GenericConfig` based on the doc string.

Reply:
I mean it technically is "unprocessed"

But I'm okay with generic